python/lsst/sphgeom/_continue_class.py

```diff
@@ -0,0 +1,218 @@
+    (
+        "__qualname__",
+        "__module__",
+        "__metaclass__",
+        "__dict__",
+        "__weakref__",
+        "__class__",
+        "__subclasshook__",
+        "__name__",
+        "__doc__",
+    )
+)
+
+
+def _isAttributeSafeToTransfer(name: str, value: typing.Any) -> bool:
+    if name.startswith("__") and (
+        value is getattr(object, name, None) or name in _INTRINSIC_SPECIAL_ATTRIBUTES
+    ):
+        return False
+    return True
+
+
+def _continueClass(cls):
+    orig = getattr(sys.modules[cls.__module__], cls.__name__)
+    for name in dir(cls):
+        # Common descriptors like classmethod and staticmethod can only be
+        # accessed without invoking their magic if we use __dict__; if we use
+        # getattr on those we'll get e.g. a bound method instance on the dummy
+        # class rather than a classmethod instance we can put on the target
+        # class.
+        attr = cls.__dict__.get(name, None) or getattr(cls, name)
+        if _isAttributeSafeToTransfer(name, attr):
+            setattr(orig, name, attr)
+    return orig
+
+
+def _inf_to_limit(value: float, min: float, max: float) -> float:
+    """Map a value to a fixed range if infinite."""
+    if not math.isinf(value):
+        return value
+    if value > 0.0:
+        return max
+    return min
+
+
+def _inf_to_lat(lat: float) -> float:
+    """Map latitude +Inf to +90 and -Inf to -90 degrees."""
+    return _inf_to_limit(lat, -90.0, 90.0)
+
+
+def _inf_to_lon(lat: float) -> float:
+    """Map longitude +Inf to +360 and -Inf to 0 degrees."""
+    return _inf_to_limit(lat, 0.0, 360.0)
+
+
+@_continueClass
+class Region:
+    """A minimal interface for 2-dimensional regions on the unit sphere."""
+
+    @classmethod
+    def from_ivoa_pos(cls, pos: str) -> Region:
+        """Create a Region from an IVOA POS string.
+
+        Parameters
+        ----------
+        pos : `str`
+            A string using the IVOA SIAv2 POS syntax.
+
+        Returns
+        -------
+        region : `Region`
+            A region equivalent to the POS string.
+
+        Notes
+        -----
+        See
+        https://ivoa.net/documents/SIA/20151223/REC-SIA-2.0-20151223.html#toc12
+        for a description of the POS parameter but in summary the options are:
+
+        * ``CIRCLE <longitude> <latitude> <radius>``
+        * ``RANGE <longitude1> <longitude2> <latitude1> <latitude2>``
+        * ``POLYGON <longitude1> <latitude1> ... (at least 3 pairs)``
+
+        Units are degrees in all coordinates.
+        """
+        shape, *coordinates = pos.split()
+        coordinates = tuple(float(c) for c in coordinates)
+        n_floats = len(coordinates)
+        if shape == "CIRCLE":
+            if n_floats != 3:
+                raise ValueError(f"CIRCLE requires 3 numbers but got {n_floats} in '{pos}'.")
+            center = LonLat.fromDegrees(coordinates[0], coordinates[1])
+            radius = Angle.fromDegrees(coordinates[2])
```

Comment:
I'm wondering now whether at least `lat` for center could be specified as `+/-Inf`. The standard is terribly underspecified.

Reply:
They don't seem to have any examples that use Inf anywhere in CIRCLE so I will ignore for now.